torch/jit/_script.py

```diff
@@ -342,15 +343,19 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
     Called by ``torch.package.PackageImporter``'s Pickler's ``persistent_load`` function.
     Performs work of loading and returning a ScriptModule from a ``torch.package`` archive.
     """
-    if not isinstance(importer.zip_reader, torch._C.PyTorchFileReader):
+
+    # if 'storage_context' not in importer.external_registry:
+    #     importer.external_registry['storage_context'] = torch._C.DeserializationStorageContext()
+
+    if not isinstance(importer.zip_reader, TorchScriptPackageZipFileReader):
```

Comment:
can we make it so TorchScriptPackageZipFileReader calls this method instead of the other way around? That would make it easier for us to split package out from core and cleans up the dependency hierarchy